### PR TITLE
os: Updating host OS variants

### PR DESCRIPTION
**Pull request by garethtdavies**
Lots of updates required on [this page](https://www.balena.io/docs/reference/OS/overview/2.x/) and here is the first batch (up to BalenaOS Components) which touches a number of existing open issues.

Change-type: patch
Connects-to: #1314
Connects-to: #1016
Connects-to: #1129
Connects-to: #1091
Signed-off-by: Gareth Davies <gareth@balena.io>

### CODE REVIEW

File: shared/overview/2.x.md [hunk 1]
```diff
@@ -4,52 +4,55 @@
...
 
 We look forward to working with the community to grow and mature {{ $names.os.lower }} into an operating system with even broader device support, a broader operating envelope, and as always, taking advantage of the most modern developments in security and reliability.
 
 ## Variants of {{ $names.os.lower }}
 
-{{ $names.os.upper }} currently comes in 3 different variants all built from the same source but with slightly differing features enabled or disabled. Each version of {{ $names.os.lower }} produces the following variants:
+Each version of {{ $names.os.lower }} is available in development and production variants, both built from the same source, but with slightly differing features enabled or disabled.
```
Comment (xginn8) [inline shared/overview/2.x.md]:
```suggestion
Each version of {{ $names.os.lower }} is available in development and production variants, both built from the same source, but with slightly differing feature sets.
```

File: shared/overview/2.x.md [hunk 2]
```diff
@@ -4,52 +4,55 @@
...
-| 2.0.0+rev2.dev | standalone | This version is an unmanaged version of {{ $names.os.lower }}, it doesn't have the {{ $names.company.lower }} supervisor and does not connect to the management console. Use this variant if you want a stable linux OS to run Docker on. |
+|:----------------:|:------------:|:--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------:|
+| 2.46.1+rev3-prod | production | This is the production version of the {{ $names.company.lower }} managed OS. This is the OS you should use for any production fleet deployments |
+| 2.46.1+rev3-dev | development | This version the development version of the above and should be used when you are developing a new application and want to use the fast [local mode][local-mode] workflow. This variant should never be used in production. |
```
Comment (xginn8) [inline shared/overview/2.x.md]: GH syntax highlighting has failed here, did you confirm this link expands in the table as expected?

File: shared/overview/2.x.md [hunk 3]
```diff
@@ -4,52 +4,55 @@
...
+* Capable of entering [local mode][local-mode] for rapid development of application containers locally.
 
 __Note:__ Raspberry Pi devices don’t have Getty attached to serial.
 
-The production images have all of the above functionality disabled by default. In both forms of the OS we write logs to an 8 MB journald RAM
-buffer in order to avoid wear on the flash storage used by most of the supported boards. However, persistent logging can be enabled by setting
-the `"persistentLogging": true` key in the `config.json` file in the boot partition of the device. The logs can be accessed [via the host OS][ssh-host] at `/var/log/journal`.
+Production images disable root passwordless access, and an SSH key must be [added][config-json-ssh] to `config.json` to access a production image using a direct SSH connection. You may still access a production image by tunneling SSH through the {{ $names.company.lower }} VPN via the CLI (using `balena ssh <uuid>`) or the {{ $names.cloud.lower }} [web terminal][ssh-host]. To use SSH via the VPN, you need to have an SSH key configured on your development machine and [added][ssh-key-add] to the {{ $names.cloud.lower }} dashboard.
```
Comment (xginn8) [inline shared/overview/2.x.md]:
```suggestion
Production images disable passwordless root access, and an SSH key must be [added][config-json-ssh] to `config.json` to access a production image using a direct SSH connection. You may still access a production image by tunneling SSH through the {{ $names.company.lower }} VPN via the CLI (using `balena ssh <uuid>`) or the {{ $names.cloud.lower }} [web terminal][ssh-host]. To use SSH via the VPN, you need to have an SSH key configured on your development machine and [added][ssh-key-add] to the {{ $names.cloud.lower }} dashboard.
```

File: shared/overview/2.x.md [hunk 4]
```diff
@@ -125,26 +128,37 @@ Now for the final piece of the puzzle, the board-specific meta-{{ $names.company
...
 
 [resin]:{{ $links.mainSiteUrl }}/
-[yocto]:https://www.yoctoproject.org/
 [linux]:http://en.wikipedia.org/wiki/Linux
 [Docker]:https://www.docker.com/
 [containerisation]:http://en.wikipedia.org/wiki/Operating_system%E2%80%93level_virtualization
 [Dockerfile]:http://docs.docker.com/reference/builder/
-[deploy-to-fleet]:/learn/deploy/deployment/
-[local-mode-link]:/learn/develop/local-mode/
+[yocto]: https://www.yoctoproject.org/
```
Comment (xginn8) [inline shared/overview/2.x.md]: maybe worth sorting these since it's getting kind of long

File: shared/overview/2.x.md (hunk 2 above)
Comment (garethtdavies) [reply shared/overview/2.x.md]: Yup, it does. This table formatting isn't the greatest.

image

File: shared/overview/2.x.md (hunk 3 above)
Comment (garethtdavies) [reply shared/overview/2.x.md]: Took me about 5 mins to spot that change 😆

File: shared/overview/2.x.md (hunk 2 above)
Comment (xginn8) [reply shared/overview/2.x.md]: nice, thanks for checking that!

File: shared/overview/2.x.md [hunk 5]
```diff
@@ -125,26 +128,37 @@ Now for the final piece of the puzzle, the board-specific meta-{{ $names.company
...
-| meta-openembedded/meta-networking | https://github.com/openembedded/meta-openembedded/tree/master/meta-networking | Central point for networking-relatedpackages and configuration.                       |
-| oe-meta-go                        | https://github.com/mem/oe-meta-go                                             | OpenEmbedded layer for the Go programming language                                    |
-| poky/meta-yocto                   | https://git.yoctoproject.org/cgit/cgit.cgi/meta-yocto/                                   |                                                                                       |
-| poky/meta                         | https://git.yoctoproject.org/cgit/cgit.cgi/poky/                                         | Core functionality and configuration of Yocto Project                                 |
+| Layer Name                                  | Repository                                                                    | Description                                                                           |
```
Comment (garethtdavies) [inline shared/overview/2.x.md]: Looks like my editor formatted this table. I know this is hugely out of date but please don't review this section (unless you have a strong opinion on markdown table formatting)  as I'll do a separate Yocto PR.

File: shared/overview/2.x.md (hunk 3 above)
Comment (xginn8) [reply shared/overview/2.x.md]: heh, sorry :) i never understand why GH fails miserably sometimes with these simple diffs.

File: shared/overview/2.x.md [hunk 6]
```diff
@@ -4,52 +4,55 @@
...
-| 2.0.0+rev2.dev | standalone | This version is an unmanaged version of {{ $names.os.lower }}, it doesn't have the {{ $names.company.lower }} supervisor and does not connect to the management console. Use this variant if you want a stable linux OS to run Docker on. |
+|:----------------:|:------------:|:--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------:|
+| 2.46.1+rev3-prod | production | This is the production version of the {{ $names.company.lower }} managed OS. This is the OS you should use for any production fleet deployments |
+| 2.46.1+rev3-dev | development | This version the development version of the above and should be used when you are developing a new application and want to use the fast [local mode][local-mode] workflow. This variant should never be used in production. |
```
Comment (lurch) [inline shared/overview/2.x.md]: "This version the development version" ?? :wink:

File: shared/overview/2.x.md [hunk 7]
```diff
@@ -4,52 +4,55 @@
...
-| 2.0.0+rev2   | production   | This is the production version of the {{ $names.company.lower }} managed OS. This is the OS you should use for any production fleet deployments |
-| 2.0.0+rev2-dev | development | This version the development version of the above and should be used when you are developing a new application and want to use the fast [local mode][local-mode-link] workflow. This variant should never be used in production. |
-| 2.0.0+rev2.dev | standalone | This version is an unmanaged version of {{ $names.os.lower }}, it doesn't have the {{ $names.company.lower }} supervisor and does not connect to the management console. Use this variant if you want a stable linux OS to run Docker on. |
+|:----------------:|:------------:|:--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------:|
+| 2.46.1+rev3-prod | production | This is the production version of the {{ $names.company.lower }} managed OS. This is the OS you should use for any production fleet deployments |
```
Comment (lurch) [inline shared/overview/2.x.md]: This PR removes the "unmanaged version" from this table. Does that mean the "managed OS" here is / isn't relevant? :man_shrugging:

File: shared/overview/2.x.md [hunk 8]
```diff
@@ -4,52 +4,55 @@
...
 
-Both Prod and Dev variants will also allow the setting of a custom hostname via the `config.json`, just add `"hostname": "my-new-hostname"`. Your device will then broadcast (via Avahi) on the network as `my-new-hostname.local`. If you don't set a custom hostname, the device will default to `<short-UUID>.local`.
+On production images, nothing is written to tty1, on boot up you should only see the {{ $names.company.lower }} logo on the HDMI screen, and this will persist until your application code takes over the framebuffer. If you would like to replace the {{ $names.company.lower }} logo with your own custom splash logo, then you will need to replace splash/resin-logo.png file that you will find in the [first partition][parition] of the image (boot partition or `resin-boot`) with your own logo.
```
Comment (lurch) [inline shared/overview/2.x.md]: "replace splash/resin-logo.png file" -> "replace the splash/resin-logo.png file" ?

File: shared/overview/2.x.md [hunk 9]
```diff
@@ -4,52 +4,55 @@
...
 
 ### Standalone {{ $names.os.lower }}
 
-{{ $names.os.upper }} also comes in a Standalone or "Unmanaged" variant. This variant is exactly the same as the `-dev` variant but **does not** have the {{ $names.company.lower }} supervisor agent and has the {{ $names.company.lower }} VPN service disabled, so it will never connect to the {{ $names.company.lower }} management system.
+Images downloaded via the CLI (using `os download`), via [balena.io/os][balena-io-os], or [manually built via Yocto][yocto-build] are the same {{ $names.os.lower }} images as those downloaded from {{ $names.cloud.lower }} but are unconfigured, and will not connect to the {{ $names.cloud.lower }} servers, but still makes use of the Supervisor to keep the containers running. This version of {{ $names.os.lower }} is meant as an excellent way to get started with Docker containers on embedded systems, and you can read more about this at [balena.io/os][balena-io-os].
```
Comment (lurch) [inline shared/overview/2.x.md]: "still makes use of" -> "still make use of" ?

File: shared/overview/2.x.md (hunk 7 above)
Comment (garethtdavies) [reply shared/overview/2.x.md]: 👍 Thanks for this, I left this table unchanged (save for deleting that row) so I've now cleaned up the terminology here for consistency.

File: shared/overview/2.x.md [hunk 10]
```diff
@@ -42,15 +42,15 @@ To persist logs on the device, enable persistent logging via the [fleet configur
...
 
-On production images, nothing is written to tty1, on boot up you should only see the {{ $names.company.lower }} logo on the HDMI screen, and this will persist until your application code takes over the framebuffer. If you would like to replace the {{ $names.company.lower }} logo with your own custom splash logo, then you will need to replace splash/resin-logo.png file that you will find in the [first partition][parition] of the image (boot partition or `resin-boot`) with your own logo.
+On production images, nothing is written to tty1, on boot up you should only see the {{ $names.company.lower }} logo on the HDMI screen, and this will persist until your application code takes over the framebuffer. If you would like to replace the {{ $names.company.lower }} logo with your own custom splash logo, then you will need to replace the `splash/resin-logo.png` file that you will find in the [first partition][parition] of the image (boot partition or `resin-boot`) with your own logo.
```
Comment (lurch) [inline shared/overview/2.x.md]: Is "the HDMI screen" overly specific, given the wide range of devices supported by Balena? :man_shrugging:

**Pull request merged**